internal/controller/instaslice_daemonset.go

```diff
@@ -141,25 +139,13 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 			}
 
 			nodeName := os.Getenv("NODE_NAME")
-			if errUpdatingNodeCapacity := r.updateNodeCapacity(ctx, nodeName, allocations.Profile, emulatorMode, allocations.Allocationstatus); errUpdatingNodeCapacity != nil {
+			if errUpdatingNodeCapacity := r.updateNodeCapacity(ctx, nodeName, allocations.Profile, emulatorMode, string(allocations.Allocationstatus)); errUpdatingNodeCapacity != nil {
 				return ctrl.Result{Requeue: true}, nil
 			}
 			var deletePrepared string
-			var errDeletingCiorGi error
-
-			if emulatorMode == "false" {
-				deletePrepared, errDeletingCiorGi = r.cleanUpCiAndGi(ctx, allocations.PodUUID, instaslice)
-				if errDeletingCiorGi != nil {
-					// deleting CI and GI is important
-					//avoid "error" : "Invalid Argument"
-					// which means that slice was deleted in previous reconcile logic and controller is trying to
-					// delete an non-existent slice
-					if errDeletingCiorGi.Error() != "Invalid Argument" {
-						log.FromContext(ctx).Error(errDeletingCiorGi, "error deleting ci or gi for ", "pod", allocations.PodName)
-						return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
```

Comment:
cleanUpCiAndGi did not return an error so this code was never hit. I removed the error return which meant I can delete all the code from 152-162.

Reply:
I think we should not delete the code until we are certain this code path is not hit, FYI,  when I did scale tests with 3K pods I did see these issues

Reply:
I am certain that this code path was never hit.

Reply:
Linter stated that the error was always nil. So that was why I removed this entire code block.

Reply:
is the linter being tested in emulator mode?

Reply:
The reason I ask is because that is a valid code path in a real GPU scenario

Reply:
I don't think it matters for emulator mode. Linter does checks on the code.

It detected that this function never returned an error so I'm unclear on what you are talking about.

Reply:
So the static analysis is sure because we stopped returning errors from the method. 

```
			errDestroyingCi := ci.Destroy()
			if errDestroyingCi != nvml.SUCCESS {
				// should we return and retry?
				log.FromContext(ctx).Error(errDestroyingCi, "error deleting compute instance")
			}
			errDestroyingGi := gi.Destroy()
			if errDestroyingGi != nvml.SUCCESS {
				// should we return and retry?
				log.FromContext(ctx).Error(errDestroyingGi, "error deleting GPU instance")
			}
```
I will open up a new issue to figure out re-try strategy but I still vote for keeping that code, as I am certain I saw this issue.

Reply:
FYI: https://github.com/openshift/instaslice-operator/issues/112

Reply:
We have the history of the code in the repo so if you need it you can always find it.

I'd prefer removing code and reintroduce it once we know it works.